pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -61,46 +78,52 @@ func (nc *nodeConfig) Configure() error {
 }
 
 // configureNetwork configures k8s networking in the node
+// we are assuming that the WindowsVM and node objects are valid
 func (nc *nodeConfig) configureNetwork() error {
-	if nc.GetCredentials() == nil {
-		return fmt.Errorf("nil credentials for VM")
-	}
-
-	if nc.GetCredentials().GetIPAddress() == "" {
-		return fmt.Errorf("empty IP for VM: %v", nc.GetCredentials())
-	}
-
-	node, err := nc.getNode(nc.GetCredentials().GetIPAddress())
-	if err != nil {
-		return errors.Wrap(err, fmt.Sprintf("error getting node object for VM %s",
-			nc.GetCredentials().GetInstanceId()))
-	}
-
 	// Wait until the node object has the hybrid overlay subnet annotation. Otherwise the hybrid-overlay will fail to
 	// start
-	if err = nc.waitForNodeAnnotation(node.GetName(), HybridOverlaySubnet); err != nil {
+	if err := nc.waitForNodeAnnotation(HybridOverlaySubnet); err != nil {
 		return errors.Wrap(err, fmt.Sprintf("error waiting for %s node annotation for %s", HybridOverlaySubnet,
-			node.GetName()))
+			nc.node.GetName()))
 	}
 
 	// NOTE: Investigate if we need to introduce a interface wrt to the VM's networking configuration. This will
 	// become more clear with the outcome of https://issues.redhat.com/browse/WINC-343
 
 	// Configure the hybrid overlay in the Windows VM
-	if err = nc.Windows.ConfigureHybridOverlay(node.GetName()); err != nil {
-		return errors.Wrap(err, fmt.Sprintf("error configuring hybrid overlay for %s", node.GetName()))
+	if err := nc.Windows.ConfigureHybridOverlay(nc.node.GetName()); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("error configuring hybrid overlay for %s", nc.node.GetName()))
 	}
 
 	// Wait until the node object has the hybrid overlay MAC annotation. This is required for the CNI configuration to
 	// start.
-	if err = nc.waitForNodeAnnotation(node.GetName(), HybridOverlayMac); err != nil {
+	if err := nc.waitForNodeAnnotation(HybridOverlayMac); err != nil {
 		return errors.Wrap(err, fmt.Sprintf("error waiting for %s node annotation for %s", HybridOverlayMac,
-			node.GetName()))
+			nc.node.GetName()))
 	}
-
 	return nil
 }
 
+// getNode returns a pointer to the node object associated with the external IP provided
+func (nc *nodeConfig) getNode() error {
+	instanceID := nc.GetCredentials().GetInstanceId()
+	nodes, err := nc.k8sclientset.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "could not get list of nodes")
+	}
+	if len(nodes.Items) == 0 {
+		return fmt.Errorf("no nodes found")
+	}
+
+	for _, node := range nodes.Items {
+		if instanceID == getInstanceIDfromProviderID(node.Spec.ProviderID) {
+			nc.node = &node
+			return nil
+		}
+	}
+	return errors.Errorf("unable to find node for instanceID %s", instanceID)
+}
+
```

Comment:
Why are you moving code around unnecessarily?

Reply:
wrt https://github.com/openshift/windows-machine-config-operator/pull/36#discussion_r408389472

Reply:
was mislead by https://github.com/openshift/windows-machine-config-operator/pull/36#discussion_r408389472

Reply:
In https://github.com/openshift/windows-machine-config-operator/pull/36#discussion_r408389472 I was asking for the variable `instanceID` to be defined closer to the scope of its usage. It had nothing to do with moving the function. You still need to move the variable closer to the usage scope.